Crashlytics/UnitTests/FABURLBuilderTests.m

```diff
@@ -34,7 +34,7 @@ - (void)setUp {
 - (void)tearDown {
   [super tearDown];
 }
-
+#if !TARGET_OS_WATCH
```

Comment:
Skip this unit test for watchOS. It has a different encoding strategy and it is broken on my local watchOS unit tests.
https://github.com/firebase/firebase-ios-sdk/blob/17cc9723c8fdf469f89f65be70aca7902ec4e4f0/Crashlytics/Shared/FIRCLSNetworking/FIRCLSURLBuilder.m#L52 
<img width="667" alt="Screen Shot 2022-11-18 at 11 52 48 PM" src="https://user-images.githubusercontent.com/16548721/202834797-846fba40-3216-41ca-8710-0bd837cec676.png">

Reply:
I wonder if this is an Xcode 14 regression like #10010

Reply:
Seems like this is a more of an issue on how paths are percent encoded in `NSString `: https://stackoverflow.com/questions/44381108/inconsistencies-in-url-encoding-methods-across-objective-c-and-swift
I agree with the highest scored answer there `It is attempting to percent encode a URL as if it was just a path`, but as the comment we have I am not sure who and why we use NSCharacterSet.URLPathAllowedCharacterSet for watchOS. Let me remove the environment condition and rerun the CI/CD again see if it can pass the unit test on SPM.

Reply:
@paulb777 It failed :(
https://github.com/firebase/firebase-ios-sdk/actions/runs/3516805883/jobs/5893825977
<img width="1294" alt="Screen Shot 2022-11-21 at 12 54 54 PM" src="https://user-images.githubusercontent.com/16548721/203126460-4b8f5a39-17b8-410d-9d42-d9af201e9939.png">

Reply:
Disable this unit test for watchOS now. I'll sync up with @samedson later once he comes back from OOO.